### PR TITLE
add Apollo config center mock test

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

## What is the purpose of the change

add Apollo config center mock test

## Brief changelog

[Test] add Apollo config center mock test

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [x] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [x] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-dependencies-bom/pom.xml [hunk 1]
```diff
@@ -609,6 +609,13 @@
                 <artifactId>nacos-client</artifactId>
                 <version>${nacos_version}</version>
             </dependency>
+            <!--for dubbo-configcenter-apollo-->
+            <dependency>
+                <groupId>com.ctrip.framework.apollo</groupId>
+                <artifactId>apollo-mockserver</artifactId>
+                <version>${apollo_client_version}</version>
+                <scope>test</scope>
```
Comment (beiwei30) [inline dubbo-dependencies-bom/pom.xml]: pls. limit the test scope library to the apollo module, since this bom will be distributed to all dubbo users.

**Comment by slievrly** [comment]
> @slievrly pls. check my comments.

@beiwei30 I found some dependencies which scope defined as test in dubbo-dependencies-bom，do I also need to move them?

image

**Comment by beiwei30** [comment]
> > @slievrly pls. check my comments.
> 
> @beiwei30 I found some dependencies which scope defined as test in dubbo-dependencies-bom，do I also need to move them?

Correct, in my opinion, they should be removed too since the end user should not rely on the libraries used for unit test.

**Comment by slievrly** [comment]
@beiwei30  In fact, I want to keep apollo-client and apollo-mockserver in the same version. The apollo_client_version property is defined in dubbo-dependencies-bom. If apollo-mockserver is placed in the apollo module, this property will not be accessible.

**Comment by slievrly** [comment]
@beiwei30 already move apollo-mockserver dependency to apollo module.

**Pull request merged**